scripts/irods/test/test_resource_types.py

```diff
@@ -77,17 +77,21 @@ def tearDown(self):
 
     @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "local filesystem check")
     def test_ichksum_no_file_modified_under_replication__4099(self):
+        #
+        # TODO IS THIS TEST NEEDED ANYMORE? WHAT DOES THIS TEST MEAN?
+        #
+
```

Comment:
Do we need this test?

This test does not appear to be useful because the checksum API detects the situation and returns an error. Notice the changes to the test. The reason why this test worked before was because the checksum API read the entire file on disk rather than the data size in the catalog.

Reply:
Makes sense to me.

@alanking wrote those a couple years ago - thoughts?   Anything we're missing?

Reply:
The test has to do with triggering `fileModified` after performing a checksum which was not valid at the time of writing the test (see #4099).

The test does the following:
1. Start with 2 identical replicas under a replication resource.
2. Corrupt the data of one of the replicas.
3. Perform a checksum update on the corrupted replica. In the past, this would forcibly update the checksum in the catalog. It seems that this is not what would happen anymore due to file size mismatch (more on this below).
4. Ensure that the other replica was not overwritten by replication triggered by `fileModified`. In other words, make sure a replication did not occur due to the checksum calculation as it is not a valid trigger for `fileModified`.

If the checksum isn't being updated due to file size on disk not matching what is recorded in the catalog, then perhaps the test should corrupt the file in such a way as to match the file size on disk with what is recorded in the catalog so that the checksum calculation goes through.

Would that restore the validity of the test? Happy to discuss whether this is a valid test, but that's the idea behind it.

Reply:
I'll adjust how much data is written on the overwrite. That should restore its validity.

Reply:
Overwriting the file with enough data allows the test to pass. I think it is good now.